### ISSUE TITLE
Learning rate scheduler have different APIs

### ISSUE THREAD

**Issue by ddkang** [pytorch/pytorch#2849]
The default API for step is either step() or step(epoch): https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L26

The default API for ReduceLROnPlateau is step(metric):
https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L252

Moreover, **this is not documented**. In fact, none of the LR schedulers have .step documented. Please either document this thoroughly or fix it.
http://pytorch.org/docs/master/optim.html#how-to-adjust-learning-rate

**Comment by DavidNemeskey** [comment]
Not to mention that for `ReduceLROnPlateau`, `step` should probably be called at the end of the epoch (after validation), and for the rest, at the beginning.

**Comment by Rikorose** [comment]
Also `ReduceLROnPlateau` has no `state_dict()` and `load_state_dict` attribute which is useful to store and resume training.

cc @zou3519
